tests/compiler/test_basics.py

```diff
@@ -427,3 +429,62 @@ def test_unpack_null_dictionary(snippetcompiler):
             "std::replace(hello_world,**dct) ({dir}/main.cf:4))"
         ),
     )
+
+
+@pytest.mark.parametrize_any(
+    "deprecated, replaced_by",
+    [(True, None), (True, '"new_function"'), (False, None), (False, '"new_function"')],
+)
+def test_modules_plugin_deprecation(
```

Comment:
Is there another way to test this than by creating a project and compiling it?

Reply:
v1 modules are a lot faster, so that would help .

Reply:
We could create a plugin and call it with plain Python. But given how much the compiler interfaces with this (it triggers the loading of Python files, it collects all registered plugins, ...) I think we get a lot more confidence out of an actual end-to-end test like this.